pkg/artifacts/artifact_storage_test.go

```diff
@@ -165,6 +256,132 @@ func TestInitializeArtifactStorageWithConfigMap(t *testing.T) {
 	}
 }
 
+func TestCleanupArtifactStorage(t *testing.T) {
+	logger := logtesting.TestLogger(t)
+	for _, c := range []struct {
+		desc        string
+		configMap   *corev1.ConfigMap
+		pipelinerun *v1alpha1.PipelineRun
+		hasPVC      bool
+	}{{
+		desc: "valid bucket",
+		configMap: &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: system.GetNamespace(),
+				Name:      v1alpha1.BucketConfigName,
+			},
+			Data: map[string]string{
+				v1alpha1.BucketLocationKey:              "gs://fake-bucket",
+				v1alpha1.BucketServiceAccountSecretName: "secret1",
+				v1alpha1.BucketServiceAccountSecretKey:  "sakey",
+			},
+		},
+		pipelinerun: &v1alpha1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "foo",
+				Name:      "pipelineruntest",
+			},
+		},
+		hasPVC: false,
+	}, {
+		desc: "location empty",
+		configMap: &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: system.GetNamespace(),
+				Name:      v1alpha1.BucketConfigName,
+			},
+			Data: map[string]string{
+				v1alpha1.BucketLocationKey:              "",
+				v1alpha1.BucketServiceAccountSecretName: "secret1",
+				v1alpha1.BucketServiceAccountSecretKey:  "sakey",
+			},
+		},
+		pipelinerun: &v1alpha1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "foo",
+				Name:      "pipelineruntest",
+			},
+		},
+		hasPVC: true,
+	}, {
+		desc: "missing location",
+		configMap: &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: system.GetNamespace(),
+				Name:      v1alpha1.BucketConfigName,
+			},
+			Data: map[string]string{
+				v1alpha1.BucketServiceAccountSecretName: "secret1",
+				v1alpha1.BucketServiceAccountSecretKey:  "sakey",
+			},
+		},
+		pipelinerun: &v1alpha1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "foo",
+				Name:      "pipelineruntest",
+			},
+		},
+		hasPVC: true,
+	}, {
+		desc: "no config map data",
+		configMap: &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: system.GetNamespace(),
+				Name:      v1alpha1.BucketConfigName,
+			},
+		},
+		pipelinerun: &v1alpha1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "foo",
+				Name:      "pipelineruntest",
+			},
+		},
+		hasPVC: true,
+	}, {
+		desc: "no secret",
+		configMap: &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: system.GetNamespace(),
+				Name:      v1alpha1.BucketConfigName,
+			},
+			Data: map[string]string{
+				v1alpha1.BucketLocationKey: "gs://fake-bucket",
+			},
+		},
+		pipelinerun: &v1alpha1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "foo",
+				Name:      "pipelineruntest",
+			},
+		},
+		hasPVC: false,
+	}} {
+		t.Run(c.desc, func(t *testing.T) {
+			fakekubeclient := fakek8s.NewSimpleClientset(c.configMap)
+
+			if c.hasPVC {
+				fakekubeclient = fakek8s.NewSimpleClientset(c.configMap, GetPVCSpec(c.pipelinerun))
```

Comment:
maybe we shouldn't call the line above in this case (i.e. move line above to `else`?)

Reply:
Weirdly, I can't figure out how to get it to work with `fakekubeclient` as a `var` - can't get the typing to match up for some reason.